.circleci/config.yml

```diff
@@ -0,0 +1,137 @@
+version: 2.1
+
+executors:
+  ruby:
+    parameters:
+      version:
+        description: "Ruby version number"
+        default: "3.1.0"
+        type: string
+    docker:
+      - image: cimg/ruby:<< parameters.version >>
+  ruby_legacy:
+    parameters:
+      version:
+        description: "Ruby version number"
+        default: "2.4.10"
+        type: string
+    docker:
+      - image: circleci/ruby:<< parameters.version >>
+
+commands:
+  bundle_install:
+    description: Install Ruby dependencies with Bundler
+    parameters:
+      version:
+        description: "Ruby version number"
+        default: "3.1.0"
+        type: string
+      bundler:
+        description: "Bundler version number"
+        default: "2.3.4"
+        type: string
+    steps:
+      - restore_cache:
+          keys:
+            - bundle-v1-{{ arch }}-<< parameters.version >>
+      - run:
+          name: Install Ruby Dependencies
+          command: |
+            gem install bundler -v '<< parameters.bundler >>' --conservative --no-document
+            bundle config --local path vendor/bundle
+            bundle check || (bundle install --jobs=4 --retry=3 && bundle clean)
+      - save_cache:
+          paths:
+            - ./vendor/bundle
+          key: bundle-v1-{{ arch }}-<< parameters.version >>-{{ checksum "Gemfile.lock" }}
+
+jobs:
+  danger:
+    executor: ruby
+    steps:
+      - checkout
+      - bundle_install
+      - run: bundle exec danger
+
+  rubocop:
+    executor: ruby
+    steps:
+      - checkout
+      - bundle_install
+      - run: bundle exec rubocop
+
+  spec:
+    parameters:
+      version:
+        description: "Ruby version number"
+        default: "3.1.0"
+        type: string
+    executor:
+      name: ruby
+      version: << parameters.version >>
+    steps:
+      - checkout
+      - bundle_install:
+          version: << parameters.version >>
+      - run: bundle exec rake spec
+
+  spec_legacy:
+    parameters:
+      version:
+        description: "Ruby version number"
+        default: "2.4.10"
+        type: string
+    executor:
+      name: ruby_legacy
+      version: << parameters.version >>
+    steps:
+      - checkout
+      - bundle_install:
+          version: << parameters.version >>
+          bundler: "< 2"
+      - run: bundle exec rake spec
+
+  spec_2_0:
+    docker:
+      - image: ruby:2.0.0-p648
```

Comment:
Circle doesn't have any docker image for Ruby 2.0. So to test 2.0 we have to break the pattern, bypass the Circle convenience images, and use the `ruby` image directly.